sdk/python/foundation-models/system/inference/fill-mask/fill-mask-batch-endpoint.ipynb

```diff
@@ -0,0 +1,483 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Prepare data for inference.\n",
+    "\n",
+    "A copy of the book corpus dataset is available in the [book-corpus-dataset](./book-corpus-dataset/) folder. The next few cells show basic data preparation:\n",
+    "* Visualize some data rows.\n",
+    "* We will `<mask>` one work in each sentence so that the model can predict the masked words.\n",
+    "* We want this sample to run quickly, so save a smaller dataset containing a fraction of the original.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define directories and filenames as variables\n",
+    "dataset_dir = \"book-corpus-dataset\"\n",
+    "training_datafile = \"train.jsonl\"\n",
+    "\n",
+    "batch_input_file = \"batch_input.csv\"\n",
+    "batch_dir = os.path.join(dataset_dir, \"batch\")\n",
+    "os.makedirs(batch_dir, exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the ./book-corpus-dataset/train.jsonl file into a pandas dataframe and show the first 5 rows\n",
+    "pd.set_option('display.max_colwidth', 0) # set the max column width to 0 to display the full text\n",
+    "train_df = pd.read_json(os.path.join(\".\", dataset_dir, training_datafile), lines=True)\n",
+    "train_df.head()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Transform the data using the masking token."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get the right mask token from huggingface\n",
+    "with urllib.request.urlopen(f\"https://huggingface.co/api/models/{model_name}\") as url:\n",
+    "    data = json.load(url)\n",
+    "    mask_token = data[\"mask_token\"]\n",
+    "\n",
+    "# Take the value of the \"text\" column, replace a random word with the mask token, and save the result in the \"masked_text\" column\n",
+    "train_df[\"masked_text\"] = train_df[\"text\"].apply(lambda x: x.replace(random.choice(x.split()), mask_token, 1))\n",
+    "\n",
+    "# Save the train_df dataframe to a jsonl file in the ./book-corpus-dataset folder with the `masked_` prefix\n",
+    "masked_datafile = os.path.join(\".\", dataset_dir, \"masked_\" + training_datafile)\n",
+    "train_df.to_json(masked_datafile, orient=\"records\", lines=True)\n",
+    "train_df.head()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Save a tenth of the input data to a file for testing batch inference. The MLflow model's signature specifies the input should be a column named `\"input_string\"`, so rename the transformed `\"masked_text\"` column. "
```

Comment:
The sampling fraction below is 0.01. Should we say 1% here or update the sampling fraction to 0.1?

```suggestion
    "Save one percent of the input data to a file for testing batch inference. The MLflow model's signature specifies the input should be a column named `\"input_string\"`, so rename the transformed `\"masked_text\"` column. "
```

Reply:
made 'a fraction' to keep it generic.